jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -647,4 +649,28 @@ private void writeTarFile(
   private <E> List<Future<E>> scheduleCallables(ImmutableList<? extends Callable<E>> callables) {
     return callables.stream().map(executorService::submit).collect(Collectors.toList());
   }
+
+  private String computeArchitecture(String architecture) {
+    if (architecture.equals("x86_64")) {
+      return "amd64";
+    } else if (architecture.equals("aarch64")) {
+      return "arm64";
+    }
+    return architecture;
+  }
+
+  private Image fetchBaseImageForLocalBuild(DockerClient dockerClient)
+      throws IOException, InterruptedException, ExecutionException {
+    DockerInfoDetails dockerInfoDetails = dockerClient.info();
+    String osType = dockerInfoDetails.getOsType();
+    String dockerArchitecture = computeArchitecture(dockerInfoDetails.getArchitecture());
+    for (Map.Entry<Image, Future<Image>> imageEntry :
+        results.baseImagesAndBuiltImages.get().entrySet()) {
+      Image image = imageEntry.getValue().get();
+      if (image.getArchitecture().equals(dockerArchitecture) && image.getOs().equals(osType)) {
+        return image;
+      }
+    }
+    return results.baseImagesAndBuiltImages.get().values().iterator().next().get();
```

Comment:
Shall we error out if no matching found?

Reply:
Yup, good idea, done.